### PR TITLE
Add heap checksums to gpinitsystem ref and GUC reference

### PR DESCRIPTION
**Pull request by litzell**
Adds HEAP_CHECKSUM to the gpinitsystem config file section and the heap_checksums GUC to the GUC reference.

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 1]
```diff
@@ -5479,6 +5480,33 @@
       </table>
     </body>
   </topic>
+  <topic id="heap_checksums">
+    <title>heap_checksums</title>
+    <body>
+      <p>Reports whether heap checksums are enabled in the database system. </p>
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: Might want to mention
-- heap checksum is set at system init. and is on by default and cannot be changed after init.
-- AO checksum is controlled at the table level and is on by default.

Also an xref to the gucs that turn heap/AO checksum errors to warnings when you get it.

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 2]
```diff
@@ -467,6 +467,7 @@
             <li>
               <xref href="#gpperfmon_port"/>
             </li>
+            <li><xref href="#heap_checksums" format="dita"/></li>
```
Comment (Chibin) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: Might be a bit confusing but, the `heap_checksums` guc is actually `data_checksums`.

File: gpdb-doc/dita/utility_guide/admin_utilities/gpinitsystem.xml [hunk 3]
```diff
@@ -472,6 +472,22 @@ declare -a PRIMARY_ARRAY=(
...
+                        databases, enabling Greenplum Database to detect corruption in the I/O
+                        system. This option is set when the system is initialized and cannot be
+                        changed later.</pd>
+                    <pd>The <codeph>HEAP_CHECKSUM</codeph> option is on by default and turning it
+                        off is strongly discouraged. If you set this option to off, data corruption
+                        in storage can go undetected and make recovery much more difficult.</pd>
+                    <pd>To determine if heap checksums are enabled in a Greenplum Database system,
+                        you can query the <codeph>heap_checksums</codeph> server configuration
+                        parameter with the <codeph>gpconfig</codeph> management
+                        utility:<codeblock>$ gpconfig -s heap_checksums</codeblock>
```
Comment (Chibin) [inline gpdb-doc/dita/utility_guide/admin_utilities/gpinitsystem.xml]: Should be `gpconfig -s data_checksums`

File: gpdb-doc/dita/utility_guide/admin_utilities/gpinitsystem.xml [hunk 4]
```diff
@@ -472,6 +472,22 @@ declare -a PRIMARY_ARRAY=(
                         and content IDs to match the values of an existing Greenplum Database
                         backup.</pd>
                 </plentry>
+                <plentry>
+                    <pt>HEAP_CHECKSUM</pt>
+                    <pd><b>Optional.</b> This parameter specifies if checksums are enabled for heap
+                        data. When enabled, checksums are calculated for heap storage in all
+                        databases, enabling Greenplum Database to detect corruption in the I/O
+                        system. This option is set when the system is initialized and cannot be
```
Comment (Chibin) [inline gpdb-doc/dita/utility_guide/admin_utilities/gpinitsystem.xml]: I wonder if there's a way to express the importance of not being able to change this in a later time.

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 5]
```diff
@@ -1294,6 +1295,33 @@
       </table>
     </body>
   </topic>
+  <topic id="heap_checksums">
+    <title>data_checksums</title>
+    <body>
+      <p>Reports whether heap checksums are enabled in the database system. </p>
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: Might want to mention that heap checksums are for heap user tables, catalog tables, and heap database metadata on master, primary and mirror segments.

Might want to check w/ C&S about how to phrase that.

Comment (asimrp) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: One way to phrase that would be: "heap checksums protect heap user tables and catalog tables on master, primary and mirror segments"

**Pull request merged**